### PR TITLE
Graph for total profit

### PR DESCRIPTION
**Pull request by CarstenWalther**
The benchmark script now generates a graph of the total profit.

Do you want a graph for total revenue?
We need a new kafka topic for that.

Example:
profit

### CODE REVIEW

File: helper_scripts/analyze.py [hunk 1]
```diff
@@ -50,8 +50,12 @@ def analyze_kafka_dump(directory):
...
 
     create_inventory_graph(directory, merchant_id_mapping)
-    create_profit_per_minute_graph(directory, merchant_id_mapping)
-    create_revenue_per_minute_graph(directory, merchant_id_mapping)
+    create_line_chart(directory, merchant_id_mapping,
+        topic='profitPerMinute', value_name='profit', label='Profit per Minute')
+    create_line_chart(directory, merchant_id_mapping,
+        topic='revenuePerMinute', value_name='revenue', label='Revenue per Minute')
+    create_line_chart(directory, merchant_id_mapping,
+        topic='profit', value_name='profit', label='Profit')
```
Comment (Bouncner) [inline helper_scripts/analyze.py]: Can you name `cumulativeProfit` or something similar? With `profit` I would be expecting a simple bar chart of the final profit and not profit over time.

**Comment by CarstenWalther** [comment]
I renamed the label to 'Cumulative Profit'.
Adding the cumulative revenue topic shouldn't take too long, but I need to touch a Flink job.

File: helper_scripts/analyze.py [hunk 2]
```diff
@@ -50,8 +50,12 @@ def analyze_kafka_dump(directory):
                              order_cost[merchant_id], profit[merchant_id]])
 
     create_inventory_graph(directory, merchant_id_mapping)
-    create_profit_per_minute_graph(directory, merchant_id_mapping)
-    create_revenue_per_minute_graph(directory, merchant_id_mapping)
+    create_line_chart(directory, merchant_id_mapping,
+        topic='profitPerMinute', value_name='profit', label='Profit per Minute')
+    create_line_chart(directory, merchant_id_mapping,
+        topic='revenuePerMinute', value_name='revenue', label='Revenue per Minute')
```
[the remark this answers is not in the document]
Comment (Bouncner) [reply helper_scripts/analyze.py]: Please also change the file name to something like `cumulative_profit.png` or `cumulativeProfit.png`).

[the remark this answers is not in the document]
Comment (Bouncner) [reply helper_scripts/analyze.py]: Aaaand one more thing, could you please unify the X-axis?
Right now, I would vote for going full length: `2018-07-06 15:03:20`.

**Comment by CarstenWalther** [comment]
If you merge the Flink Job first, I could add the revenue graph in this PR.

File: helper_scripts/analyze.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (CarstenWalther) [reply helper_scripts/analyze.py]: I will add the file names explicitly. Do we want png or svg?

**Comment by CarstenWalther** [comment]
* filenames are consistent
* x-axis shows the complete date
The images are larger on the left side to fit the larger labels.
cumulative_profit

File: helper_scripts/analyze.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Bouncner) [reply helper_scripts/analyze.py]: As of now, we only use png. But why generating both?

What do you think?

**Comment by Bouncner** [comment]
Ok, looks good. Can I merge?

**Comment by Bouncner** [comment]
I get the following error (I've pull and rebuilt the analytics image):
```
Traceback (most recent call last):
  File "/home/Martin.Boissier/pricewars/helper_scripts/benchmark.py", line 178, in <module>
    main()
  File "/home/Martin.Boissier/pricewars/helper_scripts/benchmark.py", line 174, in main
    analyze_kafka_dump(output_dir)
  File "/home/Martin.Boissier/pricewars/helper_scripts/analyze.py", line 62, in analyze_kafka_dump
    topic='cumulativeRevenue', value_name='revenue', label='Cumulative Revenue', filename='cumulative_revenue.png')
  File "/home/Martin.Boissier/pricewars/helper_scripts/analyze.py", line 76, in create_chart
    events = json.load(open(os.path.join(directory, 'kafka', topic)))
FileNotFoundError: [Errno 2] No such file or directory: '/home/Martin.Boissier/results/2018-07-06T22-24-41/kafka/cumulativeRevenue'
```

File: helper_scripts/analyze.py [hunk 3]
```diff
@@ -49,9 +49,18 @@ def analyze_kafka_dump(directory):
...
+        topic='inventory_level', value_name='level', label='Inventory Level',
+        filename='inventory_levels.png', drawstyle='steps-post')
+    create_chart(directory, merchant_id_mapping,
+        topic='profitPerMinute', value_name='profit', label='Profit per Minute', filename='profit_per_minute.png')
+    create_chart(directory, merchant_id_mapping,
+        topic='revenuePerMinute', value_name='revenue', label='Revenue per Minute', filename='revenue_per_minute.png')
+    create_chart(directory, merchant_id_mapping,
+        topic='profit', value_name='profit', label='Cumulative Profit', filename='cumulative_profit.png')
+    create_chart(directory, merchant_id_mapping,
+        topic='cumulativeRevenue', value_name='revenue', label='Cumulative Revenue', filename='cumulative_revenue.png')
```
Comment (Bouncner) [inline helper_scripts/analyze.py]: Please see my comment from 7th of July.

Comment (Bouncner) [reply helper_scripts/analyze.py]: Ping @CarstenWalther  :)

Comment (CarstenWalther) [reply helper_scripts/analyze.py]: Now the graph is skipped if a topic file is missing. This kafka topic is new. Are you on the newest version of the `analytics` subrepo.

Comment (Bouncner) [reply helper_scripts/analyze.py]: Yes, I have been using the latest analytics master commit Anyhow ... merging now.

**Pull request merged**